packages/stylelint-plugin/src/rules/no-prefix-literal.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2021 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Root, Result } from "postcss";
+import parser from "postcss-selector-parser";
+import stylelint from "stylelint";
+import type { Plugin, RuleTesterContext } from "stylelint";
+
+import { checkImportExists } from "../utils/checkImportExists";
+import { insertImport } from "../utils/insertImport";
+
+const ruleName = "@blueprintjs/no-prefix-literal";
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+    expected: (unfixed: string, fixed: string) => `Use the \`${fixed}\` variable instead of the \`${unfixed}\` literal`,
+});
+
+const bannedPrefixes = ["bp1", "bp2", "bp3", "bp4"];
+
+interface Options {
+    disableFix?: boolean;
+    variablesImportPath?: Partial<Record<Exclude<CssSyntax, CssSyntax.OTHER>, string>>;
+}
+
+// eslint-disable-next-line import/no-default-export
+export default stylelint.createPlugin(ruleName, ((
+    enabled: boolean,
+    options: Options | undefined,
+    context: RuleTesterContext,
+) => (root: Root, result: Result) => {
+    if (!enabled) {
+        return;
+    }
+
+    const validOptions = stylelint.utils.validateOptions(
+        result,
+        ruleName,
+        {
+            actual: enabled,
+            optional: false,
+            possible: [true, false],
+        },
+        {
+            actual: options,
+            optional: true,
+            possible: {
+                disableFix: [true, false],
+                variablesImportPath: (obj: unknown) => {
+                    if (typeof obj !== "object" || obj == null) {
+                        return false;
+                    }
+                    // Check that the keys and their values are correct
+                    const allowedKeys = new Set<string>(Object.values(CssSyntax).filter(v => v !== CssSyntax.OTHER));
+                    return Object.keys(obj).every(key => allowedKeys.has(key) && typeof (obj as any)[key] === "string");
+                },
+            },
+        },
+    );
+
+    if (!validOptions) {
+        return;
```

Comment:
if the options are invalid, does this mean it will silently fail?

Reply:
VS code will display an error in the bottom right corner. This is also the pattern that all stylelint plugins that I found use